### ISSUE TITLE
`saveTransaction` outside try-catch block kills optimist

### ISSUE THREAD

**Issue by druiz0992** [EYBlockchain/nightfall_3#1048]
When rebooting an optimist with a non empty Db,  it dies when attempting to resync

![Screenshot from 2022-10-05 23-34-49.png](https://images.zenhubusercontent.com/5caa6d91ae36b90f32ba4620/95eb5254-63ec-4808-bbcf-b96d9446abef).

Optimist is resyncing, and when it saves the transaction in line 73, there is a Transaction Replay error. This error is caught in the catch segment. But in line 83 there is a new attempt to save the transaction, and this time without protection that kills Optimist.

Does this work?

```  
try {
    transaction = await checkAlreadyInBlock(transaction);
    // if transaction is already in a block, its already saved
    if (fromBlockProposer) {
      saveTransaction({ ...transaction }); 
    }
  
    await checkTransaction(transaction, true);
    logger.info('Transaction checks passed');

    if (!fromBlockProposer) {
      saveTransaction({ ...transaction });
    } 
  } catch (err) {
    if (err instanceof TransactionError) {
      logger.warn(
        `The transaction check failed with error: ${err.message}. The transaction has been ignored`,
      );
    } else {
      logger.error(err);
    }
  }

**Comment by Westlad** [comment]
As part of this issue, can we expand the optimist sync test so that it also tests with a full db as well as an empty one? That would catch such problems in future.

**Comment by Westlad** [comment]
Closing as it's addressed by merging #1049
